bundle/src/main/java/com/adobe/acs/commons/indesign/dynamicdeckdynamo/services/impl/XMLGeneratorServiceImpl.java

```diff
@@ -0,0 +1,301 @@
+                    DynamicDeckUtils.getOrCreateFolder(resourceResolver, DamConstants.SUBASSETS_FOLDER, deckResource));
+            if (null != subassetsFolderResource) {
+                AssetManager assetManager = resourceResolver.adaptTo(AssetManager.class);
+                if (assetManager == null) {
+                    LOGGER.error("Asset manager is null");
+                    return null;
+                }
+                processXmlAsset = assetManager.createAsset(
+                        subassetsFolderResource.getPath() + DynamicDeckDynamoConstants.SLASH + xmlName, resultInputStream,
+                        DynamicDeckDynamoConstants.XML_MIME_TYPE, true);
+                LOGGER.debug("XML stored at {}", processXmlAsset.getPath());
+            } else {
+                LOGGER.debug("Asset subfolder is null, where processed xml asset needs to be created {}",
+                        deckResource.getPath());
+            }
+        } catch (IOException | ParserConfigurationException | SAXException | TransformerException e) {
+            LOGGER.error("Error while generation of xml.", e);
+        } finally {
+            if (!targetFile.delete()) {
+                throw new DynamicDeckDynamoException("Temporary file cannot be deleted or it's null");
+            }
+        }
+        return processXmlAsset == null ? null : processXmlAsset.getPath();
+    }
+
+    private void populateDocNodes(Document doc, List<XMLResourceIterator> assetItrList, Resource masterResource, List<String> imageList) {
+        populateHeaderValues(doc, masterResource, imageList);
+        populateIterableValues(doc, assetItrList, imageList);
+    }
+
+    private void populateIterableValues(Document doc, List<XMLResourceIterator> assetItrList, List<String> imageList) {
+        NodeList documentChildren = doc.getChildNodes();
+        List<Node> iterableSections = new ArrayList<>();
+        findSections(documentChildren, DynamicDeckDynamoConstants.XML_SECTION_TYPE_ITERABLE, iterableSections);
+        Map<String, List<Node>> iterableSectionTypes = new HashMap<>();
+        segregateSections(iterableSections, iterableSectionTypes);
+        for (XMLResourceIterator assetItr : assetItrList) {
+            String itrType = assetItr.getKey();
+            if (iterableSectionTypes.containsKey(itrType)) {
+                ListIterator<Resource> assetsIterator = assetItr.getIterator();
+                List<Node> iterablesList = iterableSectionTypes.get(itrType);
+                assetsIterator.forEachRemaining(assetResource -> {
+                    ValueMap assetProperties = assetResource.getValueMap();
+                    for (Node node : iterablesList) {
+                        duplicateNodeAndSetValues(node, assetProperties, assetResource, imageList);
+                    }
+                });
+                for (Node node : iterablesList) {
+                    node.getParentNode().removeChild(node);
```

Comment:
this is strange. What are you trying to achieve here? do you wan to delete ```node```? Why don't you use ```node.remove()``` directly?
In any case you should save the session to persist this deletion; I cannot find here, is it done anywhere else?